node/src/components/contract_runtime/operations.rs

```diff
@@ -48,47 +162,33 @@ pub(super) async fn commit_execution_effects(
             effect
         }
     };
-    let commit_result = commit(
+    let new_state_root = commit(
         engine_state,
         metrics,
         state_root_hash,
         execution_effect.transforms,
-    )
-    .await;
-    trace!(?commit_result, "commit result");
-    match commit_result {
-        Ok(CommitResult::Success { state_root }) => {
-            debug!(?state_root, "commit succeeded");
-            Ok((state_root.into(), execution_result))
-        }
-        _ => {
-            error!(
-                ?commit_result,
-                "commit failed - internal contract runtime error"
-            );
-            Err(())
-        }
-    }
+    )?;
+    Ok((new_state_root, execution_result))
 }
 
-pub(super) async fn commit(
-    engine_state: Arc<EngineState<LmdbGlobalState>>,
-    metrics: Arc<ContractRuntimeMetrics>,
+fn commit(
+    engine_state: &EngineState<LmdbGlobalState>,
+    metrics: &ContractRuntimeMetrics,
     state_root_hash: Digest,
     effects: AdditiveMap<Key, Transform>,
-) -> Result<CommitResult, engine_state::Error> {
+) -> Result<Digest, engine_state::Error> {
     trace!(?state_root_hash, ?effects, "commit");
     let correlation_id = CorrelationId::new();
     let start = Instant::now();
     let result = engine_state.apply_effect(correlation_id, state_root_hash.into(), effects);
```

Comment:
So `apply_effect` already writes the result to the database?

Reply:
Yes.